Lib/multiprocessing/pool.py

```diff
@@ -219,6 +232,23 @@ def __init__(self, processes=None, initializer=None, initargs=(),
                   self._result_handler, self._cache),
             exitpriority=15
             )
+        self._state = RUN
+
+    def __del__(self, _warn=warnings.warn):
+        if self._state == RUN:
+            _warn(f"unclosed running multiprocessing pool {self!r}", ResourceWarning,
+                  source=self)
+        elif self._state != INIT and not self._joined:
+            _warn(f"multiprocessing pool {self!r} was not joined",
+                  ResourceWarning,
+                  source=self)
+
+    def __repr__(self):
+        cls = self.__class__
+        state = _STATES.get(self._state, self._state)
+        return (f'<{cls.__module__}.{cls.__qualname__} '
+               f'state={state} '
+               f'pool#={len(self._pool)}>')
```

Comment:
Is this "#" on purpose?

Reply:
It's on purpose. '#' means 'size' or 'count': it's the number of items (processes or threads) in the pool. It's not the right syntax?

Reply:
I understand now. I think I was confused about searching for some other meaning for the '#'.

Reply:
I can remove '#' if you prefer, I don't really care :-) Or rename to 'pool_size' maybe?

Reply:
I am happy with the '#' :)

Reply:
I'd rather have "pool size".